### DISCUSSION TITLE
How to run Terrariaserver.exe with config file?

### DISCUSSION THREAD

**Discussion by Jawbone220** [Pryaxis/TShock#1933] [Q&A] [answered]
Am coming from Linux where I could run ./TerrariaServer.bin.x86_64 -config serverconfig.txt 

How do I do this natively on windows with TShock? The server runs fine but I am trying to automate it selecting the correct world and setting the basic parameters like player number, port etc. Was also looking for a startserver.bat as the Wiki says but didn't find it. 

Then how can I automate windows10 running this each time on boot so server starts each time?

Thanks!

**Comment by Jawbone220** [comment]
Alternatively, I also have docker for windows if this would be easier to run it in a container (still trying to figure that part out, however. Just installed docker engine last night)

**Comment by kubedzero** [comment, marked as answer]
startserver.bat exists in the default terraria installation (at least when downloaded via steam)

https://terraria.gamepedia.com/Server#Command_line_parameters here's information about params that could be passed in via a .bat file or a shortcut.

My config starts the server with a particular world on a certain port. The below text exists in a .bat file, and I just execute the batch file any time I want to run the server. `C:\Users\User\Desktop\tshock1.4.0.4pre8\TerrariaServer.exe -port 7778 -world "C:\Users\User\My Documents\My Games\Terraria\Worlds\WorldName.wld" -killinactivesocket`

**Reply by Jawbone220** [reply]
Thank you! I wrote the batch how you have yours and it works! Two questions - 
1. I am trying to add the password cmd line and it isn't working I've tried several different ways. Is there some trick or certain order this should be in? I have tried -pass <pw> and -pass <pw> / -password <pw>; both in different orders in the sequence, to no avail.

2. How can I have win10 start this batch file on boot? Just add to startup folder?

**Reply by kubedzero** [reply]
1. Password is not set via command line but rather in the file `config.json` within the tshock folder that lives in the same directory as TerrariaServer.exe. It's called "ServerPassword" and is set to "" by default (no password) so you can just add something in there and then the password will be set. 
2. The way I usually do it is to create a shortcut to the .bat file and then put it in the `C:\Users\User\AppData\Roaming\Microsoft\Windows\Start Menu\Programs\Startup` folder

**Reply by Jawbone220** [reply]
Awesome, thank you! I appreciate your help.

**Reply by Jonth4th** [reply]
so I need to rename `TerrariaServer.exe` to `TerrariaSrrver.exe -port 7778` if want the port to be 7778? @kubedzero

**Reply by sgkoishi** [reply]
> so I need to rename `TerrariaServer.exe` to `TerrariaSrrver.exe -port 7778` if want the port to be 7778? @kubedzero

@Jonth4th You don't rename. It's called [command-line argument](https://en.wikipedia.org/wiki/Command-line_interface#Arguments).

**Reply by Jonth4th** [reply]
> @Jonth4th You don't rename. It's called [command-line argument](https://en.wikipedia.org/wiki/Command-line_interface#Arguments).

then where do I write it? in `serverconfig.txt`?
I've done that but if I start `TerrariaServer.exe` it just starts with no configurations.

**Reply by sgkoishi** [reply]
> > @Jonth4th You don't rename. It's called [command-line argument](https://en.wikipedia.org/wiki/Command-line_interface#Arguments).
> 
> then where do I write it? in `serverconfig.txt`? I've done that but if I start `TerrariaServer.exe` it just starts with no configurations.

[Google it](https://www.google.com/search?q=where+to+write+command+line+arguments). Also note that there is no `TerrariaServer`, the executable was called `TShock.Server` since TShock v5.x.
